bookkeeper-server/src/main/java/org/apache/bookkeeper/client/ReadOnlyLedgerHandle.java

```diff
@@ -326,11 +326,15 @@ CompletableFuture<Versioned<LedgerMetadata>> closeRecovered() {
                     return builder.withClosedState().withLastEntryId(lac).withLength(len).build();
                 },
                 this::setLedgerMetadata).run();
-        f.thenRun(() -> {
-                synchronized (metadataLock) {
-                    newEnsemblesFromRecovery.clear();
-                }
-            });
+        f.whenComplete((result, exception) -> {
+            synchronized (metadataLock) {
+                newEnsemblesFromRecovery.clear();
+            }
+            if (exception != null) {
+                LOG.error("When closeRecovered,failed on clearing newEnsemblesFromRecovery: {}",
+                        BKException.getExceptionCode(exception));
```

Comment:
Get exception code is only a number? Why not dumping the whole exception (with stacktrace)

Reply:
+1

Reply:
good suggestion, I have updated
@hangc0276 @eolivelli